frontend/awx/resources/templates/WorkflowVisualizer/hooks/useRemoveAllNodes.tsx

```diff
@@ -0,0 +1,59 @@
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useBulkConfirmation, ITableColumn, TextCell } from '../../../../../../framework';
+import { action, useVisualizationController } from '@patternfly/react-topology';
+import type { GraphNode } from '../types';
+
+export function useRemoveAllNodes() {
+  const { t } = useTranslation();
+
+  const controller = useVisualizationController();
+  const handleRemoveAllNodes = action(() => {
+    controller.getElements().forEach((element) => {
+      controller.removeElement(element);
+    });
+  });
+
+  const nameColumn = useMemo<ITableColumn<GraphNode>>(
+    () => ({
+      header: t('Name'),
+      cell: (item: GraphNode) => <TextCell text={item.getLabel() || t('DELETED')} iconSize="sm" />,
+    }),
+    [t]
+  );
+  const idColumn = useMemo<ITableColumn<GraphNode>>(
+    () => ({
+      header: t('Identifier'),
```

Comment:
The identifier becomes that same value as "alias" - which is a field in the node form - if the user adds a value there. In the case where they do not, the identifier comes from the API and may not be of much use for the user.  

To help us with this we do have a util function called `stringIsUUID`

Perhaps in this delete modal we should have a alias/idenifier column, a resource name column..... what do you think?

Reply:
@marshmalien and I spoke offline and my concerns have been addressed.  I agree with this approach.